subprojects/docs/src/docs/userguide/customTasks.adoc

```diff
@@ -250,6 +250,108 @@ Using Gradle's `IncrementalTaskInputs` is not the only way to create tasks that
 
 However, if the state files are non-relocatable, then they can't be shared via the build cache. Indeed, when the task is loaded from cache, any such state files must be cleaned up to prevent stale state to confuse the tool during the next execution. Gradle can ensure such stale files are removed if they are declared via `task.localState.register()` or a property is marked with the `@LocalState` annotation.
 
+[[sec:declaring_and_using_command_line_options]]
+=== Declaring and Using Command Line Options
+
+[NOTE]
+====
+The API for exposing command line options is an <<feature_lifecycle,incubating>> feature.
+====
+
+A custom task implementation exposes properties to make the runtime behavior configurable. Sometimes a user wants to declare the value of an exposed task property on the command line instead of the build script. That's particular helpful if the property value changes more frequently. The task API supports a mechanism for marking a property to automatically generate a corresponding command line parameter with a specific name at runtime.
+
+[[sec:declaring_task_option]]
+==== Declaring an option
+
+Exposing a new command line parameter for a task property is straightforward. You just have to annotate the corresponding setter method of a property with api:org.gradle.api.tasks.options.Option[]. An option requires a mandatory identifier. Additionally, you can provide an optional description. A task can expose as many command line options as properties available in the class.
+
+Let's have a look at an example to illustrate the functionality. The custom task `UrlVerify` verifies whether a given URL can be resolved by making a HTTP call and checking the response code. The URL to be verified is configurable through the property `url`. The setter method for the property is annotated with api:org.gradle.api.tasks.options.Option[].
+
+++++
+<sample id="stringBasedOption" dir="userguide/tasks/commandLineOption/stringOption/buildSrc/src/main/java" title="Declaring a command line option">
+    <sourcefile file="UrlVerify.java" snippet="custom-task-implementation"/>
+</sample>
+++++
+
+All options declared for a task can be <sec:listing_task_options,rendered as console output>> by running the `help` task.
```

Comment:
Does it show with `gradle help` or do you have to run `gradle help --task myTask`? Let's link to the section below or just inline `gradle help --task myTask`.

Reply:
The options for a task are only being rendered with the `--task` option. Just running `help` renders this:

```
:help

Welcome to Gradle 4.6-20180116070000+0000.

To run a build, run gradle <task> ...

To see a list of available tasks, run gradle tasks

To see a list of command-line options, run gradle --help

To see more detail about a task, run gradle help --task <task>
```

I change the text to also mention `--task`. The link already exist.